server/src/Dao/RevisionDao.php

```diff
@@ -17,23 +19,18 @@ class RevisionDao extends AbstractDao {
 	public function getUserRevisionsInCommonPages(
 		$users, $startDate = null, $endDate = null, $limit = 50, $continue = null
 	) {
-		// build subquery to find common pages between users
-		$subQuery = $this->conn->createQueryBuilder();
-		$subQuery->select( 'rev_page' )
-			->from( 'revision_userindex' )
-			->where( 'rev_user_text in (:users)' )
-			->groupBy( 'rev_page' )
-			->having( 'count(distinct rev_user) > 1' );
+		$pages = $this->getUsersCommonPages( $users );
 
 		// build query to find interaction between users in common pages
 		$query = $this->conn->createQueryBuilder();
 		$query->select( 'rev_id' )
 			->from( 'revision_userindex', 'r' )
-			->where( 'rev_user_text in (:users)' )
-			->andWhere( $query->expr()->in( 'rev_page', $subQuery->getSQL() ) )
+			->where( 'rev_user in (:users)' )
+			->andWhere( 'rev_page in (:pages)' )
```

Comment:
I imagine the SQL for this query is gigantic?

Reply:
It is, but in this case it is faster than using the subquery.